core/src/test/java/kafka/clients/consumer/AsyncKafkaConsumerIntegrationTest.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.kafka.common.utils.LogCaptureAppender;
+
+import org.apache.log4j.Level;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@ExtendWith(ClusterTestExtensions.class)
+public class AsyncKafkaConsumerIntegrationTest {
+    private ClusterInstance clusterInstance;
+
+    @ClusterTest(serverProperties = {
+            @ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "1"),
+            @ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1"),
+            @ClusterConfigProperty(key = "group.coordinator.new.enable", value = "true"),
+            @ClusterConfigProperty(key = "group.coordinator.rebalance.protocols", value = "classic")
+    })
+    public void testRebalanceProtocolWithoutConsumer(ClusterInstance clusterInstance) throws Exception {
+        this.clusterInstance = clusterInstance;
+        String topic = "test-topic";
+        createTopic(topic);
+        try (LogCaptureAppender appender = LogCaptureAppender.createAndRegister()) {
+            appender.setClassLogger(KafkaConsumer.class, Level.ERROR);
+            KafkaConsumer<String, String> consumer = createAsyncKafkaConsumer("test-group");
+            consumer.subscribe(Collections.singletonList(topic));
+            TestUtils.waitForCondition(() -> {
+                try {
+                    consumer.poll(Duration.ofMillis(1000));
+                    return false;
+                } catch (UnsupportedVersionException e) {
+                    return true;
+                }
+            }, "Should get UnsupportedVersionException cause of Unsupported API");
+            TestUtils.waitForCondition(() -> appender.getEvents().stream()
+                            .filter(e -> e.getLevel().equals(Level.ERROR.toString()))
+                            .anyMatch(e -> e.getMessage().contains("The cluster doesn't yet support the new consumer group protocol. " +
+                                    "Set group.protocol=classic to revert to the classic protocol until the cluster is upgraded.")),
+                    "Error message should contain how to revert to classic protocol");
+            consumer.close();
+        }
+    }
+
+    @ClusterTest(serverProperties = {
+            @ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "1"),
+            @ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1"),
+            @ClusterConfigProperty(key = "group.coordinator.new.enable", value = "false")
+    })
+    public void testDisableNewGroupCoordiantor(ClusterInstance clusterInstance) throws Exception {
+        this.clusterInstance = clusterInstance;
+        String topic = "test-topic";
+        createTopic(topic);
+        try (LogCaptureAppender appender = LogCaptureAppender.createAndRegister()) {
```

Comment:
The body is exactly the same as the previous one. Could we extract it into an helper method?